src/DefaultBuilder/test/Microsoft.AspNetCore.TrimmingTests/UseHttpsDoesDependOnX509Test.cs

```diff
@@ -0,0 +1,23 @@
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+
+builder.WebHost.UseKestrel(serverOptions =>
+{
+    serverOptions.ListenLocalhost(5000, listenOptions =>
+    {
+#pragma warning disable SYSLIB0026 // The constructor obsolete but we're not actually going to use the cert
+        listenOptions.UseHttps(new X509Certificate2());
```

Comment:
You're directly referencing the cert. The assert that the cert type is loaded doesn't make sense anymore.

Change the `GetType` to be something internal to Kestrel? Or wrap `builder.Build()` in a try-catch and comment that the error is expected because a dev cert isn't present.

Reply:
:facepalm:

In my defence, the only purpose of the test is to confirm the helper can detect a particular preserved type, which it still does.  But I like the try-catch idea.